benchmark/test/reference/spmv_distributed.profile.stderr

```diff
@@ -116,6 +116,18 @@ DEBUG: begin components::convert_idxs_to_ptrs
 DEBUG: end   components::convert_idxs_to_ptrs
 DEBUG: begin components::convert_idxs_to_ptrs
 DEBUG: end   components::convert_idxs_to_ptrs
+DEBUG: begin components::fill_array
+DEBUG: end   components::fill_array
+DEBUG: begin components::fill_array
+DEBUG: end   components::fill_array
+DEBUG: begin copy
+DEBUG: end   copy
+DEBUG: begin copy
+DEBUG: end   copy
+DEBUG: begin copy
+DEBUG: end   copy
+DEBUG: begin copy
+DEBUG: end   copy
```

Comment:
I only found the one array copy from row_gather when copy the matrix.
Do I miss anything?

Reply:
The constructor has 
- 2 `set_executor`
- 1 `make_temporary_clone(segmented_array)`

One `set_executor` might not trigger the copy section, the other operations, however, should trigger the copy section. That gives 3 sections already, since the segmented array consists of two arrays internally. These two might also cover the fill_array section.

Reply:
The reference test should not need any copy in this three function, right?

Reply:
I found the issue. The `CollectiveCommunicator::create_with_same_type` takes in a `const std::variant<imap<...>...>&`, but this does *not* mean that the `imap` is also passed as a `const &`. Instead this triggers the copy constructor of the `imap`, which leads to the additional calls.
See this example: https://godbolt.org/z/r1WYKv37q

I will try to see if I can prevent this somehow.

Reply:
I changed the `create_with_same_type` a bit now so that it doesn't create temporary index maps. So the overhead should be gone.